pkg/splunk/enterprise/secrets.go

```diff
@@ -240,7 +240,12 @@ func ApplyNamespaceScopedSecretObject(client splcommon.ControllerClient, namespa
 	namespacedName := types.NamespacedName{Namespace: namespace, Name: GetNamespaceScopedSecretName(namespace)}
 	err := client.Get(context.TODO(), namespacedName, &current)
 	if err == nil {
-		// Found, generate values for only missing types of tokens them
+		// Found, if data is non-existent, create a map
+		if current.Data == nil {
+			current.Data = make(map[string][]byte)
```

Comment:
What if we move this line to line 238 and remove this `if` statement? It should work right or Get will still make it null if it doesn't find anything ?

Reply:
Addressed